kasa/smart/modules/firmware.py

```diff
@@ -62,21 +62,23 @@ def __init__(self, device: SmartDevice, module: str):
                     type=FeatureType.Switch,
                 )
             )
-        self._add_feature(
-            Feature(
-                device,
-                "Update available",
-                container=self,
-                attribute_getter="update_available",
-                type=FeatureType.BinarySensor,
+        if device._is_cloud_connected:
+            self._add_feature(
```

Comment:
We could expose this always to keep it simple.

Reply:
How do we make sure it shows as Unavailable in HA? Is it enough to simply return None?

Reply:
It is not currently implemented, but I suppose design-wise we could set `_attr_available` based on Noneness, much like we consider `None`s in the library as unavailable?

Reply:
Done